presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -368,12 +401,44 @@ public Type getFinalType()
         @Override
         public Type getIntermediateType()
         {
-            throw new UnsupportedOperationException();
+            return new ArrayType(RowType.anonymous(spillingTypes));
         }
 
         @Override
         public void addInput(GroupByIdBlock groupIdsBlock, Page page)
         {
+            if (page.getBlock(page.getChannelCount() - 1) instanceof ArrayBlock) {
```

Comment:
We need a better way to tell if unspill is happening, if the distinct aggregation isnt the last variable in the select block this will just fail.